security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -201,3 +218,19 @@ func (c *CitadelClient) reconnectIfNeeded() error {
 	citadelClientLog.Errorf("recreated connection")
 	return nil
 }
+
+func mapToStruct(metadata map[string]interface{}) (*types.Struct, error) {
```

Comment:
would it make sense to add a test for this function? Also, is this functionality not available in any of the current Istio packages?

Reply:
remvoed this function now --consturct the struct directly according to the suggestion from John